ext/vulnsrc/alpine/alpine.go

```diff
@@ -195,42 +132,59 @@ type secDBFile struct {
 	} `yaml:"packages"`
 }
 
-func parseYAML(r io.Reader) (vulns []database.VulnerabilityWithAffected, err error) {
-	var rBytes []byte
-	rBytes, err = ioutil.ReadAll(r)
+func parseSecDBFile(filePath string) (file *secDBFile, err error) {
+	var f io.ReadCloser
+	f, err = os.Open(filePath)
 	if err != nil {
 		return
 	}
 
-	var file secDBFile
-	err = yaml.Unmarshal(rBytes, &file)
+	defer f.Close()
+	data, err := ioutil.ReadAll(f)
 	if err != nil {
 		return
 	}
 
-	for _, pack := range file.Packages {
-		pkg := pack.Pkg
-		for version, vulnStrs := range pkg.Fixes {
-			err := versionfmt.Valid(dpkg.ParserName, version)
-			if err != nil {
-				log.WithError(err).WithField("version", version).Warning("could not parse package version. skipping")
+	file = &secDBFile{}
+	err = yaml.Unmarshal(data, file)
+	return
+}
+
+func extractVulnerabilities(file *secDBFile) (vulns []database.VulnerabilityWithAffected) {
```

Comment:
This looks like a good place to make this a method and turn parseSecDBFile into a constructor:

secDBFile => secDB
parseSecDBFile => newSecDB
extractVulnerabilites => secDB.Vulnerabilities()